src/diffTool/jsonDiff.ts

```diff
@@ -0,0 +1,383 @@
+  let typedDiff;
+  const diffNode = diff[0];
+  const id = diffNode[JSON_LD_KEY_ID];
+  const diffType = getDiffType(diff);
+  ramlToolLogger.debug(
+    `Parsing differences of node: ${id}, difference type: ${DiffType[diffType]}`
+  );
+  switch (diffType) {
+    case DiffType.ADDED:
+      typedDiff = new NodeDiff(id);
+      typedDiff.added[id] = diffNode;
+      break;
+    case DiffType.REMOVED:
+      typedDiff = new NodeDiff(id);
+      typedDiff.removed[id] = diffNode;
+      break;
+    case DiffType.MOVED:
+      ramlToolLogger.debug(`Ignoring the move of node: ${id}`);
+      break;
+    default:
+      /**
+       *  Note: MODIFIED, TEXT_DIFF are not valid for node as diff is by node id
+       */
+      logError(
+        `Invalid difference type for node: ${DiffType[diffType]}, node: ${id}`
+      );
+  }
+  return typedDiff;
+}
+
+/**
+ * Add node property differences to NodeDiff object
+ * @param key Name of the property that has difference
+ * @param diff Differences of the property
+ * @param diffType Type of the difference
+ * @param typedDiff Type object for the difference
+ * @param isArray Flag to indicate if the property is a array
+ */
+function addNodePropertyDiffs(
+  key: string,
+  diff: Array<any>,
+  diffType: DiffType,
+  typedDiff: NodeDiff
+): void {
+  switch (diffType) {
+    case DiffType.ADDED:
+      typedDiff.added[key] = diff[0];
+      break;
+    case DiffType.MODIFIED:
+      typedDiff.removed[key] = diff[0];
+      typedDiff.added[key] = diff[1];
+      break;
+    case DiffType.REMOVED:
+      typedDiff.removed[key] = diff[0];
+      break;
+    case DiffType.TEXT_DIFF:
+      //TODO: Handle text diffs where the text length exceeds the configured length
+      //parseTextDiff(diff[0);
+      break;
+    default:
+      logError(
+        `Invalid difference type for node property: ${DiffType[diffType]},  node: ${typedDiff.id}, property: ${key}`
+      );
+  }
+}
+
+function addNodeArrayPropertyDiffs(
+  key: string,
+  diff: Array<any>,
+  diffType: DiffType,
+  typedDiff: NodeDiff
+): void {
+  switch (diffType) {
+    case DiffType.ADDED:
+      if (typedDiff.added[key] == null) {
+        typedDiff.added[key] = [];
+      }
+      typedDiff.added[key].push(diff[0]);
+      break;
+    case DiffType.REMOVED:
+      if (typedDiff.removed[key] == null) {
+        typedDiff.removed[key] = [];
+      }
+      typedDiff.removed[key].push(diff[0]);
+      break;
+    case DiffType.MOVED:
+      ramlToolLogger.debug(
+        `Ignoring the moves of node: ${typedDiff.id}, property: ${key}`
+      );
+      break;
+    default:
+      logError(
+        `Invalid difference type for node array property value: ${DiffType[diffType]},  node: ${typedDiff.id}, property: ${key}`
+      );
+  }
+}
+
+/**
+ * jsondiffpatch plugin to add node ID to the difference
+ */
+// eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+const addDiffId = <any>function(context) {
```

Comment:
Why is `<any>` here?

Reply:
This is an interface in jsondiffpatch with a bare function and additional parameter. This is the simplest way I found to implement  it